test/services/platform-environment-requirements.ts

```diff
@@ -0,0 +1,210 @@
+
+function createTestInjector() {
+	const testInjector = new Yok();
+
+	testInjector.register("commandsService", {currentCommandData: {commandName: "test", commandArguments: [""]}});
+	testInjector.register("doctorService", {});
+	testInjector.register("errors", {
+		fail: (err: any) => {
+			throw new Error(err.formatStr || err.message || err);
+		}
+	});
+	testInjector.register("extensibilityService", {});
+	testInjector.register("logger", stubs.LoggerStub);
+	testInjector.register("prompter", {});
+	testInjector.register("platformEnvironmentRequirements", PlatformEnvironmentRequirements);
+	testInjector.register("staticConfig", { SYS_REQUIREMENTS_LINK: "" });
+
+	return testInjector;
+}
+
+describe("platformEnvironmentRequirements ", () => {
+	describe("checkRequirements", () => {
+		let testInjector: IInjector = null;
+		let platformEnvironmentRequirements: IPlatformEnvironmentRequirements = null;
+
+		beforeEach(() => {
+			testInjector = createTestInjector();
+			platformEnvironmentRequirements = testInjector.resolve("platformEnvironmentRequirements");
+			process.stdout.isTTY = true;
+			process.stdin.isTTY = true;
+		});
+
+		it("should show prompt when environment is not configured", async () => {
+			const doctorService = testInjector.resolve("doctorService");
+			doctorService.canExecuteLocalBuild = () => false;
+
+			let isPromptForChoiceCalled = false;
+			const prompter = testInjector.resolve("prompter");
+			prompter.promptForChoice = () => {
+				isPromptForChoiceCalled = true;
+				return PlatformEnvironmentRequirements.CLOUD_BUILDS_OPTION_NAME;
+			};
+
+			let isInstallExtensionCalled = false;
+			const extensibilityService = testInjector.resolve("extensibilityService");
+			extensibilityService.installExtension = () => {isInstallExtensionCalled = true; };
+			extensibilityService.getInstalledExtensions = () => { return {}; };
+
+			await assert.isRejected(platformEnvironmentRequirements.checkEnvironmentRequirements(platform));
+			assert.isTrue(isPromptForChoiceCalled);
+			assert.isTrue(isInstallExtensionCalled);
+		});
+
+		it("should return true when environment is configured", async () => {
+			const doctorService = testInjector.resolve("doctorService");
+			doctorService.canExecuteLocalBuild = () => true;
+
+			const result = await platformEnvironmentRequirements.checkEnvironmentRequirements(platform);
+			assert.isTrue(result);
+		});
+
+		describe("when setup script option is selected ", () => {
+			it("should return true when env is configured after executing setup script", async () => {
+				let index = 0;
+				const doctorService = testInjector.resolve("doctorService");
+				doctorService.canExecuteLocalBuild = () => {
+					if (index === 0) {
```

Comment:
This logic is a bit odd IMO.
I suggest mocking `runSetupScript` like this:

`doctorService.runSetupScript = async () => { doctorService.canExecuteLocalBuild = ()=> true; }`

Then the whole code would become something like:
```ts
doctorService.canExecuteLocalBuild = () => false;
doctorService.runSetupScript = async () => { doctorService.canExecuteLocalBuild = () => true; }
```

And we'd lose the odd count of how many times `canExecuteLocalBuild` has been called